test/SerialIOTest.cpp

```diff
@@ -83,6 +83,74 @@ TEST_CASE( "multi_series_test", "[serial]" )
     allSeries.clear();
 }
 
+TEST_CASE( "available_chunks_test_json", "[serial][json]" )
+{
+    /*
+     * This test is JSON specific
+     * Our JSON backend does not store chunks explicitly,
+     * so the JSON backend will simply go through the multidimensional array
+     * and gather the data items into chunks
+     * Example dataset:
+     *
+     *      0123
```

Comment:
I think I have not yet seen a JSON file with multiple `storeChunk` files - can you please provide me with an example so I can follow a bit better? :)

Reply:
Uhm.. aren't you looking at an example? Or what would you like to see additionally?

Reply:
If it's the JSON dataset that you'd like to see:
```json
          "x": {     
            "attributes": {
              "position": {
                "datatype": "VEC_DOUBLE",
                "value": [
                  0
                ]    
              }, 
              "unitSI": {
                "datatype": "DOUBLE",
                "value": 1
              }      
            },
            "data": [                                                                                   
              [                                                                                         
                null,                                                                                   
                null,                                                                                   
                null,                                                                                   
                null                                                                                    
              ],                                                                                        
              [                                                                                         
                null,                                                                                   
                null,                                                                                   
                null,                                                                                   
                null                                                                                    
              ],                                                                                        
              [                                                                                         
                2,                                                                                      
                4,                                                                                      
                6,                                                                                      
                8                                                                                       
              ],                                                                                        
              [                                                                                         
                2,                                                                                      
                4,                                                                                      
                6,                                                                                      
                8                                                                                       
              ],                                                                                        
              [                                                                                         
                2,                                                                                      
                4,                                                                                      
                6,                                                                                      
                8
              ],
              [
                2,
                4,
                6,
                8
              ],
              [
                2,
                4,
                6,
                8
              ],
              [
                2,
                4,
                null,
                null
              ],
              [
                2,
                4,
                null,
                2
              ],
              [
                null,
                null,
                null,
                4
              ]
            ],
            "datatype": "INT"
          }
```
So, you get nulls in unwritten positions. Nulls don't get "folded up" if a whole dimension is null, since the JSON backend takes care to fill every position in the n-d dataset at least with nulls.

Reply:
Ok, that's what I am familiar with. I somehow don't get the json "chunks" implementation then. I guess there is always only one chunk... (ok)?

Reply:
Well, that would be a possibility. But in JSON, it's possible to reconstruct chunks such that exactly those positions are covered that are non-null, so I figured I'd just do that.

Reply:
Ah, got it now - smart! :tada: